tests/core/transformers/test_llm_summarize.py

```
@@ -0,0 +1,486 @@
+        # Test with failing LLM
+        mock_llm = Mock()
+        mock_llm.completion.side_effect = ConnectionError("Network timeout")
+        mock_default_llm.return_value = mock_llm
+        transformer = LLMSummarizeTransformer({"fast_model": "gpt-4o-mini"})
+
+        with pytest.raises(TransformerError) as exc_info:
+            transformer.transform("input to summarize")
+
+        # Verify error is properly wrapped and chained
+        assert "Failed to summarize content with fast model" in str(exc_info.value)
+        assert exc_info.value.__cause__.__class__ == ConnectionError
```

Ruff (0.12.2)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)